src/XIVLauncher.Common.Unix/Compatibility/DxvkSettings.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace XIVLauncher.Common.Unix.Compatibility;
+
+public class DxvkSettings
+{
+    public string DownloadURL { get; private set; }
+
+    public string FolderName { get; private set; }
+
+    public Dictionary<string, string> DxvkVars { get; private set; }
+
+    public Dxvk.DxvkHudType DxvkHud { get; private set; }
+
+    public Dxvk.DxvkVersion DxvkVersion { get; private set; }
+
+    private const string ALLOWED_CHARS = "^[0-9a-zA-Z,=.]+$";
+    private const string ALLOWED_WORDS = "^(?:devinfo|fps|frametimes|submissions|drawcalls|pipelines|descriptors|memory|gpuload|version|api|cs|compiler|samplers|scale=(?:[0-9]){0,2}(?:.(?:[0-9])+)?)$";
+
+    public DxvkSettings(Dxvk.DxvkHudType hud = Dxvk.DxvkHudType.None, string dxvkHudCustom="",
+        string mangoHudPath="", bool? async = true, int? frameRate = 0,
+        Dxvk.DxvkVersion version = Dxvk.DxvkVersion.v1_10_1)
+    {
+        this.DxvkHud = hud;
+        this.DxvkVars = new Dictionary<string, string> ();
+        this.DxvkVars.Add("DXVK_ASYNC", ((async ?? false) ? "1" : "0"));
+        frameRate ??= 0;
+        if (frameRate > 0) this.DxvkVars.Add("DXVK_FRAME_RATE", (frameRate).ToString());
+        this.DxvkVersion = version;
+        string release = this.DxvkVersion switch
+        {
+            Dxvk.DxvkVersion.v1_10_1 => "1.10.1",
+            Dxvk.DxvkVersion.v1_10_2 => "1.10.2",
+            Dxvk.DxvkVersion.v1_10_3 => "1.10.3",
+            Dxvk.DxvkVersion.v2_0 => "2.0",
+            _ => VersionOutOfRange(),
```

Comment:
This probably should be an `ArgumentOutOfRangeException`, as it is not really supposed to happen and you'd have to change the default DXVK version in the future in like 3 different places

Reply:
Alright, found a couple of examples in the code and mimicked them. It compiles, and, as you said, this shouldn't ever actually happen to a normal user. Changed.